pkg/cvo/status_test.go

```diff
@@ -190,3 +194,131 @@ func TestOperator_syncFailingStatus(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateClusterVersionStatus_UnknownVersionAndRRI(t *testing.T) {
+	ignoreLastTransitionTime := cmpopts.IgnoreFields(configv1.ClusterOperatorStatusCondition{}, "LastTransitionTime")
+
+	testCases := []struct {
+		name string
+
+		unknownVersion bool
+		oldCondition   *configv1.ClusterOperatorStatusCondition
+		failure        error
+
+		expectedRriCondition *configv1.ClusterOperatorStatusCondition
+	}{
+		{
+			name:                 "RRI disabled, version known, no failure => condition not present",
+			unknownVersion:       false,
+			expectedRriCondition: nil,
+		},
+		{
+			name:                 "RRI disabled, version known, failure => condition not present",
+			unknownVersion:       false,
+			failure:              fmt.Errorf("Something happened"),
+			expectedRriCondition: nil,
+		},
+		{
+			name: "RRI disabled, version unknown, failure => condition present",
```

Comment:
nit: maybe add "existing condition" to the name?  And a test case for `RRI disabled, version unknown, no existing condition, failure => condition not present`?

Reply:
:+1: added in 3da7615f189f94c87fc371620578dfe1e9c229cb